### PR TITLE
CB-14230 patch release 5.0.4

### PR DESCRIPTION
**Pull request by brodybits**
### Platforms affected

browser

### What does this PR do?

- _Update RELEASENOTES.md with the most important changes for `5.0.3` (was forgotten in the `5.0.3` release)_
- Update RELEASENOTES.md with _the most important_ changes for `5.0.4`
- Update JS snapshot, VERSION, and version to `5.0.4` on `5.0.x` branch using the following command with updates from apache/cordova-coho#176: `./cordova-coho/coho prepare-platform-release-branch --version 5.0.4 -r browser 5.0.x`

This release includes the following changes:
* [CB-9366](https://issues.apache.org/jira/browse/CB-9366) - log `error.stack` in `cordova.js` (update from `cordova-js@4.2.4`) in `5.0.x` (ef012e389996d27f580063a7b271f2ae08c484e9)
* add Node.js 10 to AppVeyor CI and Travis CI in `5.0.x` (57fdfc8735f6aa5b91b397f7822180ea86bdb0bf)
* [CB-14145](https://issues.apache.org/jira/browse/CB-14145) update dependencies to resolve `npm audit` issues, pinned in `5.0.x` only (GH-53)
* GH-54 - pin more added bundled dependencies in 5.0.x only (needed to clear the red highlights in `npm outdated --depth=0`)
* _GH-56 - pin even more added bundled dependencies in 5.0.x only, needed to clear a few more red highlights in `npm outdated --depth=0`_

### What testing has been done on this change?

- [x] `npm audit` shows 0 vulnerabilities (using npm@6.2.0)
- [x] `npm outdated --depth=0` shows no red entries
- [x] `coho audit-license-headers -r browser` shows no missing license headers
- [x] `coho check-license -r browser` shows no incorrect licenses _(NOTE: this patch release now includes `xmldom` package which does not have correct SPDX expression; however, visual inspection of `node_modules/xmldom/LICENSE` shows MIT license option, which is acceptable.)_
- [x] `npm test` succeeds on local machine
- [x] `npm test` succeeds on AppVeyor CI & Travis CI
- [x] able to add to <https://github.com/brodybits/cordova-dialogs-bootstrap-test> using `cordova platform add github:brodybits/cordova-browser#5.0.x`, run, and try a couple dialogs with npm@2.15.11 (bundled with deprecated node v4.9.1) _on Windows Edge, Windows Chrome, and macOS Safari_
- [x] able to add to <https://github.com/brodybits/cordova-dialogs-bootstrap-test> using `cordova platform add github:brodybits/cordova-browser#5.0.x`, run, and try a couple dialogs with npm@5.6.0 (bundled with node v8.11.3)

### Checklist

- ~~[Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database~~
- [x] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected. (manual changes only)
- ~~Added automated test coverage as appropriate for this change.~~

### CODE REVIEW

**Comment by janpio** [comment]
> Update RELEASENOTES.md with major changes for 5.0.4

major changes? Typo?

File: RELEASENOTES.md [hunk 1]
```diff
@@ -20,6 +20,21 @@
 -->
 ## Release Notes for Cordova Browser ##
 
+### 5.0.4 (Jul 30, 2018)
+* [CB-14145](https://issues.apache.org/jira/browse/CB-14145) update dependencies to resolve `npm audit` issues, pinned in `5.0.x` only ([GH-53](https://github.com/apache/cordova-browser/pull/53))
+* [CB-9366](https://issues.apache.org/jira/browse/CB-9366) - log `error.stack` in `cordova.js` (update from `cordova-js@4.2.4`) in `5.0.x`
+* [GH-53](https://github.com/apache/cordova-browser/pull/53) add Node.js 10 to AppVeyor CI and Travis CI (`5.0.x`)
+
+### 5.0.3 (Dec 20, 2017)
```
Comment (janpio) [inline RELEASENOTES.md]: Unrelated changes - should those be here?

Comment (brodybits) [reply RELEASENOTES.md]: @janpio I will assume that you mean the old `5.0.3` RELEASENOTE changes. These changes are needed somewhere since someone forgot to do them before making `5.0.3` release. I would be happy to move these to a separate PR.

**Comment by brodybits** [comment]
> major changes? Typo?

Reworded to "most important changes" in the description.

File: RELEASENOTES.md (hunk 1 above)
Comment (janpio) [reply RELEASENOTES.md]: Yes, I commented on that line exactly - Github just uselessly displays some more lines to confuse you :/

I don't really care about the separate PR, just mention it in the PR description - so no one can assume it is accidental.

Comment (brodybits) [reply RELEASENOTES.md]: > just mention it in the PR description

done (my bad)

File: RELEASENOTES.md [hunk 2]
```diff
@@ -20,6 +20,21 @@
 -->
 ## Release Notes for Cordova Browser ##
 
+### 5.0.4 (Jul 30, 2018)
+* [CB-14145](https://issues.apache.org/jira/browse/CB-14145) update dependencies to resolve `npm audit` issues, pinned in `5.0.x` only ([GH-53](https://github.com/apache/cordova-browser/pull/53))
+* [CB-9366](https://issues.apache.org/jira/browse/CB-9366) - log `error.stack` in `cordova.js` (update from `cordova-js@4.2.4`) in `5.0.x`
+* [GH-53](https://github.com/apache/cordova-browser/pull/53) add Node.js 10 to AppVeyor CI and Travis CI (`5.0.x`)
```
Comment (brodybits) [inline RELEASENOTES.md]: Adding Node.js 10 to CI was not really part of GH-53, will fix this once I resolve a few more red entries in `npm outdated --depth=0` as discussed in the description.

Comment (brodybits) [reply RELEASENOTES.md]: Now fixed, continuing with tests before pushing to `5.0.x`
